pkg/controller/periodic/job.go

```diff
@@ -29,7 +33,12 @@ func NewJobController(kubeClient kubernetes.Interface) *JobController {
 
 // CreateGathererJob creates a new Kubernetes Job with provided image, volume mount path used for storing data archives and name
 // derived from the provided data gather name
-func (j *JobController) CreateGathererJob(ctx context.Context, dataGatherName, image, archiveVolumeMountPath string) (*batchv1.Job, error) {
+func (j *JobController) CreateGathererJob(
+	ctx context.Context, dataGatherName, image string, dataReporting *config.DataReporting, storage *insightsv1alpha1.Storage,
```

Comment:
nit: we could save one argument here by passing the pointer to the full `insightsv1alpha1.DataGather`